### ISSUE TITLE
Possibility to breakdown assembly purchases

### ISSUE THREAD

**Issue by oscgonfer** [inventree/InvenTree#1192]
I came across an use case that I am not sure is supported (or if it's too much of a nuisance to implement). I drop the idea here in case you find it relevant.

Imagine you have _Parts A_, _B_ and _C_ that can be used to make an _Assembly Part X_. Also, _Parts A_, _B_ and _D_ can be used to make an _Assembly Part Y_. Let's imagine these parts are some electronic and mechanical components that can be assembled into different machines with different BOMs.

Now, let's imagine the case we purchase a _pack_ of these components, for instance, we could imagine a pack that contains some _A_, _B_ [...] _D Parts_, because a certain _supplier_ sells them that way. Is it possible to break down that purchased pack so that you can fulfill the assembly of _Parts X_ and/or _Y_?

**Comment by SchrodingersGat** [comment]
Hi @oscgonfer it's a great question. I think it is very similar to https://github.com/inventree/InvenTree/issues/1126 (would you agree)?

Currently InvenTree does not support this feature, although I would not be against supporting it. There would need to be some thought as to how it would be implemented.

**Comment by oscgonfer** [comment]
Hi @SchrodingersGat,

Absolutely, missed that issue. From my perspective (maybe this discussion could be moved to the other issue, I let you decide), at the moment of marking items as received, there could be a potentially irreversible option which breaks down the items into it's core components allowing for future builds with different BOMs.

What do you think?

**Comment by SchrodingersGat** [comment]
That's an interesting idea! An assembly can be ordered, and then disassembled based on its BOM. It is a reasonably clean approach which uses existing ideas quite nicely

**Comment by maxsimongt3** [comment]
I don't know if this helps but at the warehouse I work they do the case pack (Suplier Part)like this.
<img width="1893" alt="Screen Shot 2021-02-28 at 2 27 52 PM" src="https://user-images.githubusercontent.com/78984540/109430816-2cb33580-79d1-11eb-9d3d-a11ea80219f5.png">
when we open a case pack it launches 2 dialogue menus, the first is to insure that you want to open it.
<img width="640" alt="Screen Shot 2021-02-28 at 2 36 40 PM" src="https://user-images.githubusercontent.com/78984540/109431028-67699d80-79d2-11eb-92a4-74c3472d47ef.png">
the second is how many of them you want to open
<img width="610" alt="Screen Shot 2021-02-28 at 2 39 20 PM" src="https://user-images.githubusercontent.com/78984540/109431111-c4fdea00-79d2-11eb-97ac-2a2f4319370b.png">

when it comes to kits (Builds) it looks like this.
<img width="1868" alt="Screen Shot 2021-02-28 at 2 31 56 PM" src="https://user-images.githubusercontent.com/78984540/109431126-d47d3300-79d2-11eb-96b7-41a1b9c5da4c.png">
they can be assembled but they cannot be opened.

**Comment by oscgonfer** [comment]
hi! just checking in with you: has this been solved here https://github.com/inventree/InvenTree/pull/3644?

**Comment by SchrodingersGat** [comment]
@oscgonfer no, but check out the discussion in https://github.com/inventree/InvenTree/issues/2624 for some more info

**Comment by SchrodingersGat** [comment]
FYI - this feature is now incoming in https://github.com/inventree/InvenTree/pull/12310

**Comment by oscgonfer** [comment]
Suuuuuper!
